src/modules/dashboard/components/ExpenditurePage/Payments/LockedPayments.tsx

```diff
@@ -69,75 +77,91 @@ const LockedPayments = ({
     });
   }, []);
 
-  const displayDelay = useCallback(
-    (recipientDelay: { amount?: string; time: string } | undefined) => {
-      if (!recipientDelay) {
-        return null;
-      }
-
-      const { amount, time } = recipientDelay;
-
-      if (!amount) {
-        return null;
-      }
-
-      if (time === 'months') {
-        return amount === '1' ? `, ${amount}mth` : `, ${amount}mths`;
-      }
-      return `, ${amount}${time.slice(0, 1)}`;
-    },
-    [],
+  const pendingCreations = pendingChanges?.recipients?.filter(
+    (recipient) => recipient?.created,
   );
 
   return (
     <div className={styles.paymentContainer}>
       <div className={styles.recipientContainer}>
         <div className={styles.payments}>
           <FormattedMessage {...MSG.payments} />
+          {activeState?.id !== Stage.Claimed && (
+            <span className={styles.editIcon}>
+              <Icon
+                name="edit"
+                appearance={{ size: 'medium' }}
+                title="Edit expenditure"
+                onClick={editForm}
+              />
+            </span>
+          )}
         </div>
         {recipients?.map(({ claimed, claimDate, ...recipient }, index) => {
           const isOpen =
             expandedRecipients?.find((idx) => idx === index) !== undefined;
-          const recipientName =
-            recipient.recipient.profile.displayName ||
-            recipient.recipient.profile.username ||
-            '';
+
+          const inPendingState = pendingChanges?.recipients?.find(
+            (item) => item.id === recipient.id,
+          );
 
           return (
             <div className={styles.singleRecipient} key={recipient.id}>
```

Comment:
This is a key that is causing errors for me when I use the same recipient for multiple payments. I think we could use the count as well as the recipient.id i.e. address as the key. For example `0xaEsd0...wE73_1`.

![31620875-2a03860425fd52af46b745d2222db8e3](https://user-images.githubusercontent.com/33682027/185993354-a40b1ea8-5eca-44fd-b406-526ef725ed6e.png)